src/devices/Ds3231/Ds3231.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using System.Device.I2c.Drivers;
+using System.Runtime.InteropServices;
+
+namespace Iot.Device.Ds3231
+{
+    /// <summary>
+    /// Realtime Clock DS3231
+    /// </summary>
+    public class Ds3231 : IDisposable
+    {
+        #region Address
+        private const byte RTC_I2C_ADDR = 0x68;
+        private const byte RTC_SEC_REG_ADDR = 0x00;
+        private const byte RTC_MIN_REG_ADDR = 0x01;
+        private const byte RTC_HOUR_REG_ADDR = 0x02;
+        private const byte RTC_DAY_REG_ADDR = 0x03;
+        private const byte RTC_DATE_REG_ADDR = 0x04;
+        private const byte RTC_MONTH_REG_ADDR = 0x05;
+        private const byte RTC_YEAR_REG_ADDR = 0x06;
+        private const byte RTC_TEMP_MSB_REG_ADDR = 0x11;
+        private const byte RTC_TEMP_LSB_REG_ADDR = 0x12;
+        #endregion
+
+        private I2cDevice _sensor = null;
+
+        private readonly int _busId;
+        private readonly OSPlatform _os;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="os">The program runing platform (Linux or Windows10)</param>
+        /// <param name="busId">I2C Bus ID</param>
+        public Ds3231(OSPlatform os, int busId = 1)
+        {
+            _busId = busId;
+            _os = os;
+        }
+
+        /// <summary>
+        /// Initialize the sensor
+        /// </summary>
+        /// <returns></returns>
+        public void Initialize()
+        {
+            var settings = new I2cConnectionSettings(_busId, RTC_I2C_ADDR);
+
+            if (_os == OSPlatform.Linux)
+            {
+                _sensor = new UnixI2cDevice(settings);
+            }
+            else if (_os == OSPlatform.Windows)
+            {
+                _sensor = new Windows10I2cDevice(settings);
+            }
+        }
+
+        /// <summary>
+        /// Read Time from DS3231
+        /// </summary>
+        /// <returns>DS3231 Time</returns>
+        public DateTime ReadTime()
+        {
+            byte[] rawData = new byte[7];
+
+            _sensor.Write(new byte[] { RTC_SEC_REG_ADDR });
+            _sensor.Read(rawData);
+
+            Ds3231Data data = new Ds3231Data();
```

Comment:
Another option here would be if the constructor takes in the Span of data, and sets the properties accordingly. Bcd2Int method could be part of that class as well.